src/Microsoft.AspNet.Hosting.WindowsService/WindowsServiceWebApplication.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNet.Hosting.Internal;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System.ServiceProcess;
+
+namespace Microsoft.AspNet.Hosting.WindowsService
+{
+    public class WindowsServiceWebApplication : ServiceBase
+    {
+        private const string HostingJsonFile = "hosting.json";
+        private const string EnvironmentVariablesPrefix = "ASPNET_";
+        private const string ConfigFileKey = "config";
+        private IApplication _application;
+        private bool _stopRequestedByWindows;
+
+        public static void Run()
+        {
+            Run(new WindowsServiceWebApplication());
+        }
+
+        public static void Run(WindowsServiceWebApplication app)
```

Comment:
Is there any reason for this to be public? When I would actually call this overload from my app?